lib/open_feature/client.rb

```diff
@@ -38,12 +38,12 @@ def add_hooks(hooks)
     sig do
       params(
         flag_key: String,
-        default_value: T::Boolean,
+        default_value: T.nilable(T::Boolean),
```

Comment:
We don't want to change the top-level API for flag evaluation. [According to the spec](https://openfeature.dev/specification/sections/flag-evaluation#conditional-requirement-1311), `default_value` is a required parameter with no explicit default. 

Further, forcing the caller to think through the default for each situation seems valuable to me. While most default values for boolean evals are probably false, that seems pretty dangerous to enforce and might lead to unexpected behavior, say, if someone thinks the default _should_ be true.